plugins/processors/scale/scale_test.go

```diff
@@ -0,0 +1,283 @@
+						"test1": float64(50),
+						"test2": float64(100),
+					}, time.Unix(0, 0)),
+				testutil.MustMetric("Name2", map[string]string{},
+					map[string]interface{}{
+						"test1": float64(75),
+						"test2": float32(25),
+					}, time.Unix(0, 0)),
+				testutil.MustMetric("Name3", map[string]string{},
+					map[string]interface{}{
+						"test3": float64(4.2),
+						"test4": float64(9),
+					}, time.Unix(0, 0)),
+				testutil.MustMetric("Name4", map[string]string{},
+					map[string]interface{}{
+						"test3": float64(1),
+						"test4": float64(8.2),
+					}, time.Unix(0, 0)),
+			},
+		},
+		{
+			name: "Ignored Fileds",
+			scale: &Scale{
+				Scalings: []Scaling{
+					{
+						InMin:  -1,
+						InMax:  1,
+						OutMin: 0,
+						OutMax: 100,
+						Fields: []string{"test1", "test2"},
+					},
+				},
+			},
+			inputs: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": int64(0),
+						"test2": uint64(1),
+						"test3": int64(1),
+					}, time.Unix(0, 0)),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": float64(50),
+						"test2": float64(100),
+						"test3": int64(1),
+					}, time.Unix(0, 0)),
+			},
+		},
+		{
+			name: "Out of range tests",
+			scale: &Scale{
+				Scalings: []Scaling{
+					{
+						InMin:  -1,
+						InMax:  1,
+						OutMin: 0,
+						OutMax: 100,
+						Fields: []string{"test1", "test2"},
+					},
+				},
+			},
+			inputs: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": int64(-2),
+						"test2": uint64(2),
+					}, time.Unix(0, 0)),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": float64(-50),
+						"test2": float64(150),
+					}, time.Unix(0, 0)),
+			},
+		},
+		{
+			name: "Missing field Fileds",
+			scale: &Scale{
+				Scalings: []Scaling{
+					{
+						InMin:  -1,
+						InMax:  1,
+						OutMin: 0,
+						OutMax: 100,
+						Fields: []string{"test1", "test2"},
+					},
+				},
+			},
+			inputs: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": int64(0),
+					}, time.Unix(0, 0)),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": float64(50),
+					}, time.Unix(0, 0)),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.scale.Log = testutil.Logger{}
+
+			require.NoError(t, tt.scale.Init())
+			actual := tt.scale.Apply(tt.inputs...)
+
+			testutil.RequireMetricsEqual(t, tt.expected, actual)
+		})
+	}
+}
+
+func TestErrorCases(t *testing.T) {
+	tests := []struct {
+		name     string
+		scale    *Scale
+		inputs   []telegraf.Metric
+		expected []telegraf.Metric
```

Comment:
`inputs` and `expected` are not used. However, you could test if the error contains a certain message.